components/makeUnableToEnterSubmit.tsx

```diff
@@ -0,0 +1,62 @@
+import { useRouter } from "next/router";
+import React from "react";
+import { SubmitProps, submitPropTypes } from "remultiform/step";
+import getProcessRef from "../helpers/getProcessRef";
+import useDataValue from "../helpers/useDataValue";
+import PageSlugs from "../steps/PageSlugs";
+import Storage from "../storage/Storage";
+import { SubmitButtons } from "./SubmitButtons";
+
+export interface MakeUnableToEnterSubmitProps {
+  slug?: PageSlugs;
+  cancel?: boolean;
+  value: string;
+  afterSubmit?(): Promise<void>;
+}
+
+const useUnableToEnterSlug = ():
+  | PageSlugs.FirstFailedAttempt
+  | PageSlugs.SecondFailedAttempt => {
+  const router = useRouter();
+  const processRef = getProcessRef(router);
+
+  const firstFailedAttempt = useDataValue(
+    Storage.ProcessContext,
+    "unableToEnter",
+    processRef,
+    (values) =>
+      processRef ? values[processRef]?.firstFailedAttempt : undefined
+  );
+
+  return firstFailedAttempt.result && firstFailedAttempt.result.date
+    ? PageSlugs.SecondFailedAttempt
+    : PageSlugs.FirstFailedAttempt;
+};
+
+export const makeUnableToEnterSubmit = (
+  props: MakeUnableToEnterSubmitProps
+): React.FunctionComponent<SubmitProps & { disabled?: boolean }> => {
+  const Submit: React.FunctionComponent<
+    SubmitProps & {
+      disabled?: boolean;
+    }
+  > = ({ disabled, onSubmit }) => {
+    const buttonProps = [
+      props,
+      { slug: useUnableToEnterSlug(), value: "Unable to Enter" },
```

Comment:
Let's pull the hook up to assign to a variable. Makes it easier to spot that there's a hook.